packages/parser/src/schemas/cognito.ts

```diff
@@ -0,0 +1,522 @@
+ *
+ * @example
+ * ```json
+ * {
+ *   "version": "1",
+ *   "region": "us-east-1",
+ *   "userPoolId": "us-east-1_ABC123",
+ *   "userName": "johndoe",
+ *   "callerContext": {
+ *     "awsSdkVersion": "2.814.0",
+ *     "clientId": "client123"
+ *   },
+ *   "triggerSource": "PreSignUp_SignUp",
+ *   "request": {
+ *     "userAttributes": {
+ *       "email": "johndoe@example.com",
+ *       "name": "John Doe"
+ *     },
+ *     "validationData": null,
+ *     "clientMetadata": {
+ *       "someKey": "someValue"
+ *     }
+ *   },
+ *   "response": {
+ *     "autoConfirmUser": true,
+ *     "autoVerifyEmail": false,
+ *     "autoVerifyPhone": false
+ *   }
+ * }
+ * ```
+ * @see {@link https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-pre-sign-up.html}
+ */
+export const PreSignupTriggerSchema = z.object({
+  version: z.string(),
+  region: z.string(),
+  userPoolId: z.string(),
+  userName: z.string(),
+  callerContext: z.object({
+    awsSdkVersion: z.string(),
+    clientId: z.string(),
+  }),
+  triggerSource: z.string(),
+  request: z.object({
+    userAttributes: z.record(z.string(), z.string()),
+    validationData: z.record(z.string(), z.string()).nullable().optional(),
+    clientMetadata: z.record(z.string(), z.string()).optional(),
+    userNotFound: z.boolean().optional(),
+  }),
+  response: z
+    .object({
+      autoConfirmUser: z.boolean().optional(),
+      autoVerifyEmail: z.boolean().optional(),
+      autoVerifyPhone: z.boolean().optional(),
+    })
+    .optional(),
+});
+
+/**
+ * A zod schema for a Cognito Post-Confirmation trigger event.
+ *
+ * @see {@link https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-post-confirmation.html}
+ *
+ * @example
+ * ```json
+ * {
+ *   "request": {
+ *     "userAttributes": {
+ *       "email": "user@example.com",
+ *       "name": "John Doe"
+ *     },
+ *     "clientMetadata": {
+ *       "customKey": "customValue"
+ *     }
+ *   },
+ *   "response": {}
+ * }
+ * ```
+ */
+export const PostConfirmationTriggerSchema = z.object({
+  request: z.object({
+    userAttributes: z.record(z.string(), z.string()),
+    clientMetadata: z.record(z.string(), z.string().optional()),
+  }),
+  response: z.object({}),
+});
+
+/**
+ * A zod schema for a Cognito Pre-Authentication trigger event.
+ *
+ * @example
+ * ```json
+ * {
+ *   "request": {
+ *     "userAttributes": {
+ *       "email": "user@example.com",
+ *       "name": "John Doe"
+ *     },
+ *     "validationData": {
+ *       "someKey": "someValue"
+ *     },
+ *     "userNotFound": false
+ *   },
+ *   "response": {}
+ * }
+ * ```
+ *  * @see {@link https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-pre-authentication.html}
+ */
+export const PreAuthenticationTriggerSchema = z.object({
+  request: z.object({
+    userAttributes: z.record(z.string(), z.string()),
+    validationData: z.record(z.string(), z.string()),
+    userNotFound: z.boolean(),
```

Comment:
This can be undeined, the definition should be `userNotFound: z.boolean().optional(),`